troubleshooting/debugging.md

```diff
@@ -593,6 +595,109 @@ To work around this issue, we suggest using the following option so that system
+
+1. It uses a custom config file:
+```
+			"firelensConfiguration": {
+				"type": "fluentbit",
+				"options": {
+					"config-file-type": "s3",
+					"config-file-value": "arn:aws:s3:::yourbucket/yourdirectory/extra.conf"
+				}
+			},
+```
+2. It uses logConfiguration to configure an output:
+```
+"logConfiguration": {
+				 "logDriver":"awsfirelens",
+				 "options": {
+					"Name": "kinesis_firehose",
+					"region": "us-west-2",
+					"delivery_stream": "my-stream",
+					"retry_limit": "2"
+				}
+			}
+```
+
+Perform the follpwing steps.
+1. Create a new directory to store the configuration for this repro attempt and `cd` into it. 
+2. Create a sub-directory called `socket`. This will store the unix socket that Fluent Bit recieves container stdout/stderr logs from. 
+3. Copy the custom config file into your current working directory. In this case, we will call it `extra.conf`.
+4. Create a file called `fluent-bit.conf` in yoru current working directory.
+    a. To this file we need to add the config content that FireLens generates. Recall that this is explained in [FireLens: Under the Hood](https://aws.amazon.com/blogs/containers/under-the-hood-firelens-for-amazon-ecs-tasks/). You can directly copy and paste the inputs and filters from the [generated_by_firelens.conf](https://github.com/aws-samples/amazon-ecs-firelens-under-the-hood/blob/mainline/generated-configs/fluent-bit/generated_by_firelens.conf) example from that blog. 
+    b. Add an `@INCLUDE` statement for your custom config file in the same working directory. In this case, its `@INCLDUE extra.conf`. 
+    c. [Optional for most repros] Add the null output for TCP healthcheck logs to the `fluent-bit.conf` from [here](https://github.com/aws-samples/amazon-ecs-firelens-under-the-hood/blob/mainline/generated-configs/fluent-bit/generated_by_firelens.conf#L31). Please note that the [TCP health check is no longer recommended](https://github.com/aws-samples/amazon-ecs-firelens-examples/tree/mainline/examples/fluent-bit/health-check).
+    d. Create an `[OUTPUT]` configuration based on the `logConfiguration` `options` map for each container in the task definition. The `options` simply become key value pairs in the `[OUTPUT]` configuration.
+    e. Your configuration is now ready to test! If you have a more complicated custom configuration file, you may need to perform additional steps like adding parser files to your working directly, and updating the import paths for them in the Fluent Bit `[SERVICE]` section if your custom config had that. 
+5. Next you need to run Fluent Bit. You can use the following command, which assumes you have an AWS credentials file in `$HOME/.aws` which you want to use for credentials. Run this command in the same working directory that you put your config files in. You may need to customize it to run a different image, mount a directory to dump core files (see our core file tutorial) or add environment variables. 
```

Comment:
I personally would really benefit from a tree view of the directory structure: https://www.cyberciti.biz/faq/linux-show-directory-structure-command-line/

The instructions are clear, but I have a hard time visualizing what the resulting directory pattern would be.

Reply:
Good idea. I think I will also edit this to be for the core-file option where you also mount a cores directory.